File: javascript/parsing-csv-data/parsing-a-data-file.js
import fs from "node:fs";
import Papa from "papaparse";

const csv_data = fs.readFileSync("./data.csv", "utf8").trim();
const parsed_data = Papa.parse(csv_data).data;

console.log(`${"Last".padEnd(10)} ${"First".padEnd(10)} Salary`);
console.log("-".repeat(28));
for (const row of parsed_data) {
	/** @type {string[]} */
	const [last, first, salary] = row;

	console.log(
		`${last.padEnd(10)} ${first.padEnd(10)} $${Intl.NumberFormat().format(salary)}`,
	);
}

File: python/parsing_data/parsing_a_data_file.py
import csv


def format_row(row: list[str]):
    last_name, first_name, salary = row

    print(f"{last_name:10} {first_name:10} ${int(salary):0,}")


with open("./data.csv") as file:
    rows = csv.reader(file, delimiter=",")

    print(f"{"Last":10} {"First":10} Salary")
    print("-" * 30)
    for row in rows:
        format_row(row)
